app/src/main/java/org/dhis2/data/user/UserComponent.java

```diff
@@ -175,4 +177,7 @@
 
     @NonNull
     ScheduledEventComponent plus(ScheduledEventModule scheduledEventModule);
+
+    @NonNull
+    OptionSetComponent plus(OptionSetModule optionSetModule);
```

Comment:
I would have added the OptionSetComponent as a subcomponent of the ServerComponent or the AppComponent as it just need D2 and the ScheduleProvider.